sync/handlers/leafs_request.go

```diff
@@ -220,7 +224,19 @@ func (rb *responseBuilder) fillFromSnapshot(ctx context.Context) (bool, error) {
 	// modified since the requested root. If this assumption can be verified with
 	// range proofs and data from the trie, we can skip iterating the trie as
 	// an optimization.
-	snapKeys, snapVals, err := rb.readLeafsFromSnapshot(ctx)
+	// Since we are performing this read optimistically, we use a separate context
+	// with reduced timeout so there is enough time to read the trie if the snapshot
+	// read does not contain up-to-date data.
+	snapCtx := ctx
```

Comment:
Is `snapCtx` used anywhere? It looks like it's only assigned here and once inside the `if` block, but we don't do anything with it.

Reply:
It's used on line 239

Reply:
😞 whoops